framework/src/userobject/SideIntegralVariableUserObject.C

```diff
@@ -28,13 +28,27 @@ SideIntegralVariableUserObject::SideIntegralVariableUserObject(const InputParame
                                  Moose::VarKindType::VAR_ANY,
                                  Moose::VarFieldType::VAR_FIELD_STANDARD),
     _u(coupledValue("variable")),
-    _grad_u(coupledGradient("variable"))
+    _grad_u(coupledGradient("variable")),
+    _fv(_fv_variable)
 {
   addMooseVariableDependency(&mooseVariableField());
 }
 
 Real
 SideIntegralVariableUserObject::computeQpIntegral()
 {
-  return _u[_qp];
+#ifdef MOOSE_GLOBAL_AD_INDEXING
+  if (_fv)
+  {
+    /// We should be at the edge of the domain
+    const FaceInfo * const fi = _mesh.faceInfo(_current_elem, _current_side);
+    mooseAssert(fi, "We should have a face info");
+    if (!fi->isBoundary())
+      mooseError("SideIntegralVariableUserObject should only be used at boundaries");
```

Comment:
Let's turn this into an assert. For qp methods, I'd like for things to be as fast as possible in optimized modes.